### ISSUE TITLE
Validation of All Checks Related to Addition of Java14 Enhanced instanceof Grammar

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8391]
This issue will be the main tracker of all work related to checks impacted by merging https://github.com/checkstyle/checkstyle/pull/8401. 

## SUMMARY
Possibly affected checks, and those that should apply to this syntax, are listed below. Since we have created a new token, we will need to add this token to specific checks that do apply.  The below list was determined by grep'ing the checks directory for checks that use VARIABLE_DEF. The logic here is that since we have a new variable, PATTERN_VARIABLE_DEF, we should examine all checks that have VARIABLE_DEF as a required or acceptable token, to determine relevance to our new token. 

## AFFECTED CHECKS

```
➜  checks git:(initial-pattern-def-grammar) grep -l -H "TokenTypes.VARIABLE_DEF" **/**/*.* | sort -u                       
annotation/AnnotationLocationCheck.java
annotation/AnnotationOnSameLineCheck.java
annotation/SuppressWarningsCheck.java
coding/DeclarationOrderCheck.java
coding/EqualsAvoidNullCheck.java
coding/ExplicitInitializationCheck.java
coding/FinalLocalVariableCheck.java
coding/HiddenFieldCheck.java
coding/IllegalTypeCheck.java
coding/InnerAssignmentCheck.java
coding/MagicNumberCheck.java
coding/ModifiedControlVariableCheck.java
coding/MultipleVariableDeclarationsCheck.java
coding/RequireThisCheck.java
coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java
coding/VariableDeclarationUsageDistanceCheck.java
design/HideUtilityClassConstructorCheck.java
design/InnerTypeLastCheck.java
design/InterfaceIsTypeCheck.java
design/MutableExceptionCheck.java
design/VisibilityModifierCheck.java
FinalParametersCheck.java
imports/UnusedImportsCheck.java
indentation/BlockParentHandler.java
indentation/HandlerFactory.java
indentation/MemberDefHandler.java
javadoc/AtclauseOrderCheck.java
javadoc/JavadocStyleCheck.java
javadoc/JavadocTagInfo.java
javadoc/JavadocVariableCheck.java
metrics/JavaNCSSCheck.java
modifier/InterfaceMemberImpliedModifierCheck.java
modifier/ModifierOrderCheck.java
modifier/RedundantModifierCheck.java
naming/AbbreviationAsWordInNameCheck.java
naming/ConstantNameCheck.java
naming/LocalFinalVariableNameCheck.java
naming/LocalVariableNameCheck.java
naming/MemberNameCheck.java
naming/StaticVariableNameCheck.java
whitespace/EmptyLineSeparatorCheck.java

```
Aside from the above checks, we will also create a new check, called `BestPracticeInstanceOfWithPatternVar`

**Background Information**
From [JEP](https://openjdk.java.net/jeps/305):

The instanceof grammar is extended accordingly:

_RelationalExpression:_
     _..._
     _RelationalExpression instanceof ReferenceType_
     _RelationalExpression instanceof Pattern_

_Pattern:_
     _ReferenceType Identifier_

**Comment by nmancus1** [comment]
@romani we can close this one now, too.

**Comment by romani** [comment]
This is just umbrella issue, all sub issues are closed. And released at 8.35.
